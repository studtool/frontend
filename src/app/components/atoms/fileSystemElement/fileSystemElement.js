import React, {Component} from 'react';

export function FileSystemElement (props){
    const {
        qa = false,
        title = '',
        handleSubmit = null,
    } = props;
    return (
        <div contentEditable={true} onKeyDown={handleSubmit}>
            {props.children}
        </div>
    );
}
