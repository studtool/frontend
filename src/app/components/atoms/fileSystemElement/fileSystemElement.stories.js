import * as React from 'react';
import {storiesOf} from '@storybook/react';
import {FileSystemElement} from 'App/components/atoms/fileSystemElement/fileSystemElement';

storiesOf('File system element ', module)
    .add('default', () => (
        <FileSystemElement title='Название файла'>
            Название файла
        </FileSystemElement>
    ))
    .add('with qa', () => (
        <FileSystemElement qa={true}/>
    ))
