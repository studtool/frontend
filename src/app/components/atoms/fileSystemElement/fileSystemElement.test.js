import React from 'react';
import {FileSystemElement} from './fileSystemElement';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('file System Element test', () => {
    const stubSubmit = () => {};
    const stubChange = () => {};
    const data = {
        email__errorMessage: '',
        password__errorMessage: '',
        passwordRepeat__errorMessage: '',
    };

    const tree = renderer.create(
        <FileSystemElement
            handleSubmit={stubSubmit}
            handleChange={stubChange}
            data={data}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
