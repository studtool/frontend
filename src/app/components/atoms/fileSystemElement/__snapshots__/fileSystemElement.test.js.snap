// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file System Element test 1`] = `
<div
  contentEditable={true}
  onKeyDown={[Function]}
/>
`;
